Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationFailureRepositoryTest.cs

```diff
@@ -0,0 +1,498 @@
+
+      var result = repository.GetValidationUnhandledFailuresForBocListAndContainingDataRowsAndDataCells(false);
+      Assert.That(result, Is.EqualTo(failures));
+
+      result = repository.GetValidationUnhandledFailuresForBocListAndContainingDataRowsAndDataCells(false);
+      Assert.That(result, Is.EqualTo(failures));
+    }
+
+    [Test]
+    public void GetValidationFailuresForBocListAndContainingDataRowsAndDataCells_WithCellAndRowHandledSeparately_OnlyContainsUnhandledFailures ()
+    {
+      var repository = new BocListValidationFailureRepository();
+
+      var unmarkedRowFailure = BusinessObjectValidationFailure.Create("Unmarked row failure");
+      var markedRowFailure = BusinessObjectValidationFailure.Create("Marked row failure");
+      var unmarkedCellFailure = BusinessObjectValidationFailure.Create("Unmarked cell failure");
+      var markedHiddenCellFailure = BusinessObjectValidationFailure.Create("Hidden marked cell failure");
+      var markedCellFailure = BusinessObjectValidationFailure.Create("Marked cell failure");
+      var markedColumnStub = new Mock<IBocColumnDefinitionWithValidationSupport>();
+      var markedRowStub = new Mock<IBusinessObject>();
+
+      repository.AddValidationFailuresForDataRow(_rowObjectStub.Object, new[] { unmarkedRowFailure });
+      repository.AddValidationFailuresForDataRow(markedRowStub.Object, new[] { markedRowFailure });
+      repository.AddValidationFailuresForDataCell(markedRowStub.Object, markedColumnStub.Object, new[] { markedHiddenCellFailure });
+
+      repository.AddValidationFailuresForDataCell(_rowObjectStub.Object, _columnDefinitionStub.Object, new[] { unmarkedCellFailure });
+      repository.AddValidationFailuresForDataCell(_rowObjectStub.Object, markedColumnStub.Object, new[] { markedCellFailure });
+
+      var result = repository.GetValidationUnhandledFailuresForBocListAndContainingDataRowsAndDataCells(false);
+      Assert.That(result, Is.EquivalentTo(new[] { unmarkedCellFailure, unmarkedRowFailure, markedRowFailure, markedCellFailure, markedHiddenCellFailure }));
+
+      result = repository.GetUnhandledValidationFailuresForDataCell(_rowObjectStub.Object, markedColumnStub.Object, true);
+      Assert.That(result, Is.EqualTo(new[] { markedCellFailure }));
+
+      result = repository.GetValidationUnhandledFailuresForBocListAndContainingDataRowsAndDataCells(false);
+      Assert.That(result, Is.EquivalentTo(new[] { unmarkedCellFailure, unmarkedRowFailure, markedRowFailure, markedHiddenCellFailure }));
+
+      result = repository.GetUnhandledValidationFailuresForDataRowAndContainingDataCells(markedRowStub.Object, true);
+      Assert.That(result, Is.EquivalentTo(new[] { markedRowFailure, markedHiddenCellFailure }));
+
+      result = repository.GetValidationUnhandledFailuresForBocListAndContainingDataRowsAndDataCells(false);
+      Assert.That(result, Is.EquivalentTo(new[] { unmarkedCellFailure, unmarkedRowFailure }));
```

Comment:
I like the updated naming. It is really helping with undestanding the expected test results  without needing to reason over the entire test and test name. Basically, I can trust in that by using easily to understand expected-values, there's a much lower chance of typo-style mistakes in the expectations because you would stumble over your own naming while setting up the expectations.

"trust & review": https://blog.ploeh.dk/2023/03/20/on-trust-in-software-development/
TL;DR: everyone makes mistakes, so we do reviews to spot them